addons/Omen/Omen.lua

```diff
@@ -0,0 +1,287 @@
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+_addon.name    = 'Omen'
+_addon.author  = 'Braden, Sechs'
+_addon.version = '1.3'
+_addon.command = 'omen'
+
+config = require ('config')
+texts = require('texts')
+
+defaults = T{}
+defaults.text_R = 255 --Color values are in RGB, ranging from 0 to 255
+defaults.text_G = 255
+defaults.text_B = 255
+defaults.good_R = 0
+defaults.good_G = 255
+defaults.good_B = 0
+defaults.bad_R = 255
+defaults.bad_G = 0
+defaults.bad_B = 0
+defaults.pos_x = 0
+defaults.pos_y = 0
+defaults.font_size = 11
+
+settings = config.load(defaults)
+
+good_col = "\\cs("..tostring(settings.good_R)..","..tostring(settings.good_G)..","..tostring(settings.good_B)..")"
+bad_col = "\\cs("..tostring(settings.bad_R)..","..tostring(settings.bad_G)..","..tostring(settings.bad_B)..")"
+omens = 0
+obj_time = 0
+floor_obj = "Waiting for objectives..."
+floor_clear = ""
+IMAGE = texts.new("IMAGE")
+
+texts.color(IMAGE,settings.text_R,settings.text_G,settings.text_B)
+texts.size(IMAGE,settings.font_size)
+texts.pos_x(IMAGE,settings.pos_x)
+texts.pos_y(IMAGE,settings.pos_y)
+
+function reset_objectives()
+objectives = {
+[1] = {id=1,mes=0,amt=0,req=0},
+[2] = {id=2,mes=0,amt=0,req=0},
+[3] = {id=3,mes=0,amt=0,req=0},
+[4] = {id=4,mes=0,amt=0,req=0},
+[5] = {id=5,mes=0,amt=0,req=0},
+[6] = {id=6,mes=0,amt=0,req=0},
+[7] = {id=7,mes=0,amt=0,req=0},
+[8] = {id=8,mes=0,amt=0,req=0},
+[9] = {id=9,mes=0,amt=0,req=0},
+[10] = {id=10,mes=0,amt=0,req=0}
+}
+obj_time = 0
+floor_clear = ""
+end
+reset_objectives()
+
+function refresh()
+header = floor_clear..floor_obj.."\\cr     Omens: "..omens
+body = "\n Bonus Objectives    "..os.date('%M:%S', obj_time)
+    for k,v in pairs (hide_timer) do
+        if string.find(header,v) then
+            body = ""
+            texts.text(IMAGE,header)
+            return
+        end
+    end
+    for v=1,table.getn(objectives),1 do
+        if objectives[v].mes ~= 0 then
+            local msg = objectives[v].mes
+            local cur = objectives[v].amt
+            local fin = objectives[v].req
+            if cur == fin then
+                body = body.."\n "..good_col..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]\\cr"
+            elseif obj_time < 1 and cur < fin then
+                body = body.."\n "..bad_col..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]\\cr"
+            else
+                body = body.."\n "..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]"
+            end
+        end
+    end
+body = string.gsub(body,"%-1","%?%?%?")
+texts.text(IMAGE,header..body)
+end
+
+hide_timer = {"Kin","Gin","Kei","Kyou","Fu","Craver","Gorger","Thinker","Treasure","Waiting"}
+refresh()
+
+
+function test(str)
```

Comment:
I guess I can put em in a separate file and then custom-include it at start when I need to test. And yeah, they absolutely don't need to be there. I've been just using them to test objective tracking and colours and blahblah when outside of Omen. Good point on removing them, will do.